app/src/main/java/ru/suslovalex/androidacademyapp/FragmentMoviesList.kt

```diff
@@ -26,17 +30,29 @@ class FragmentMoviesList : Fragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        val recycler: RecyclerView = view.findViewById(R.id.rv_moviesList)
-        adapterMovies = AdapterMovies({ id -> doOnClick(id) }, getListMovies())
-        recycler.layoutManager = GridLayoutManager(view.context, 2)
-        recycler.adapter = adapterMovies
-        recycler.hasFixedSize()
+        lifecycleScope.launch(Dispatchers.IO) {
+            getMovies()
+            updateUI(view, movieList)
+        }
+    }
+
+    private suspend fun getMovies() = withContext(Dispatchers.IO){
+        movieList = loadMovies(this@FragmentMoviesList.requireContext())
     }
 
-    private fun doOnClick(id: Int) {
+    private suspend fun updateUI(view: View, movieList: List<Movie>) =
```

Comment:
@AleksandrAlex Why this fun is `suspend`, you can do this operation without coroutines